diff --git a/js/tree.js b/js/tree.js
index e49615d..4abc3fa 100644
--- a/js/tree.js
+++ b/js/tree.js
@@ -50,7 +50,7 @@
       lazyLoad: function(event, data) {
         // Load child nodes via ajax GET /taxonomy_manager/parent=1234
         data.result = {
-          url: "/taxonomy_manager/subTree",
+          url: Drupal.url("taxonomy_manager/subTree"),
           data: {parent: data.node.key},
           cache: false
         };
diff --git a/src/Element/TaxonomyManagerTree.php b/src/Element/TaxonomyManagerTree.php
index 8c4b0fe..fad8bef 100644
--- a/src/Element/TaxonomyManagerTree.php
+++ b/src/Element/TaxonomyManagerTree.php
@@ -9,7 +9,7 @@ namespace Drupal\taxonomy_manager\Element;
 
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Render\Element\FormElement;
-use Drupal\Component\Utility\HTML;
+use Drupal\Component\Utility\Html;
 
 /**
  * Taxonomy Manager Tree Form Element
@@ -143,7 +143,7 @@ class TaxonomyManagerTree extends FormElement {
     if (!empty($terms)) {
       foreach ($terms as $term) {
         $item = array(
-          'title' => HTML::escape($term->getName()),
+          'title' => Html::escape($term->getName()),
           'key' => $term->id(),
         );
 
diff --git a/src/Form/TaxonomyManagerForm.php b/src/Form/TaxonomyManagerForm.php
index 86b5366..0c45db8 100644
--- a/src/Form/TaxonomyManagerForm.php
+++ b/src/Form/TaxonomyManagerForm.php
@@ -13,7 +13,7 @@ use Drupal\Core\Ajax\ReplaceCommand;
 use Drupal\Core\Form\FormBase;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Render\Element;
-use Drupal\Component\Utility\HTML;
+use Drupal\Component\Utility\Html;
 use Drupal\taxonomy\VocabularyInterface;
 use Drupal\taxonomy_manager\TaxonomyManagerHelper;
 
@@ -102,7 +102,7 @@ class TaxonomyManagerForm extends FormBase {
 
     $form['taxonomy']['manager'] = array(
       '#type' => 'fieldset',
-      '#title' => HTML::escape($taxonomy_vocabulary->label()),
+      '#title' => Html::escape($taxonomy_vocabulary->label()),
       '#tree' => TRUE,
     );
 
